openstack/barbican/templates/barbican-nanny-deployment.yaml

```diff
@@ -0,0 +1,113 @@
+    {{ if eq .Values.pod.lifecycle.upgrades.deployments.podReplacementStrategy "RollingUpdate" }}
+    rollingUpdate:
+      maxUnavailable: {{ .Values.pod.lifecycle.upgrades.deployments.rollingUpdate.maxUnavailable }}
+      maxSurge: {{ .Values.pod.lifecycle.upgrades.deployments.rollingUpdate.maxSurge }}
+    {{ end }}
+  selector:
+    matchLabels:
+      app: barbican-nanny
+  template:
+    metadata:
+      labels:
+        app: barbican-nanny
+        name: barbican
+        alert-tier: os
+        alert-service: barbican
+        {{- include "utils.topology.pod_label" . | indent 8 }}
+      annotations:
+        chart-version: {{.Chart.Version}}
+        checksum/etc-configmap.conf: {{ include (print $.Template.BasePath "/etc-configmap.yaml") . | sha256sum }}
+        checksum/etc-configmap.conf: {{ include (print $.Template.BasePath "/scripts-configmap.yaml") . | sha256sum }}
+        secrets-hash: {{ include (print $.Template.BasePath "/secrets.yaml") . | sha256sum }}
+        {{- if .Values.proxysql.mode }}
+        prometheus.io/scrape: "true"
+        prometheus.io/targets: {{ required ".Values.alerts.prometheus missing" .Values.alerts.prometheus | quote }}
+        {{- end }}
+        {{- include "utils.linkerd.pod_and_service_annotation" . | indent 8 }}
+    spec:
+      {{ include "utils.proxysql.pod_settings" . | indent 6 }}
+{{- if .Values.barbican_nanny.db_secret_move.enabled }}
+      containers:
+      - name: move-secrets
+        image: {{required ".Values.global.registry is missing" .Values.global.registry }}/loci-barbican:{{required "Values.imageVersionBarbicanApi is missing" .Values.imageVersionBarbicanApi}}
+        imagePullPolicy: IfNotPresent
+        command:
+        - dumb-init
+        - kubernetes-entrypoint
+        env:
+        - name: COMMAND
+          value: "{{ if not .Values.barbican_nanny.debug }}/bin/bash /scripts/barbican-db-check.sh{{ else }}sleep inf{{ end }}"
+        - name: BARBICAN_DB_SECRET_MOVE_ENABLED
+          value: {{ .Values.barbican_nanny.db_secret_move.enabled | quote }}
+        - name: BARBICAN_DB_OLD_PROJECT_ID
+          value: {{ .Values.barbican_nanny.db_secret_move.old_project_id | quote }}
+        - name: BARBICAN_DB_NEW_PROJECT_ID
+          value: {{ .Values.barbican_nanny.db_secret_move.new_project_id | quote }}
+        - name: BARBICAN_NANNY_INTERVAL
+          value: {{ .Values.barbican_nanny.interval | quote }}
+        {{- if .Values.sentry.enabled }}
+        - name: SENTRY_DSN
+          valueFrom:
+            secretKeyRef:
+              name: sentry
+              key: barbican.DSN.python
+        {{- end }}
+        resources:
+          requests:
+            memory: "250Mi"
+            cpu: "25m"
+          limits:
+            memory: "250Mi"
+            cpu: "100m"
+        volumeMounts:
+        - name: barbican-scripts
+          mountPath: /scripts/barbican-db-check.sh
+          subPath: barbican-db-check.sh
```

Comment:
I propose to rename the script to match what it does and the container name: `move-secrets.sh`.